---
title: "Identifying Expression Shifts from Ancestral Gene Expression Reconstructions"
author: "Erin Fry"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    theme: cerulean
    highlight: espresso
    fig_width: 8
    fig_height: 6
    self_contained: false
---

```{r instructions and variables to define before beginning including tissue and lineages, echo=FALSE}


## Erin Fry
## July 28 2017

## this script analyzes BAGER results

## MODIFICATION INSTRUCTIONS:
## all required or recommended changes are in the first section of the script
## change the variable tissuename and tissue to the tissue name and code you are evaluating
## change the variable path to home directory for the project
## if testing other models, modify model vector to include all models
## you must modify the nodes of interest 'if then' section. Change the names of the nodes to the clade name 
## and the "Node ..." to the column names in the files created by BayesTraits for the .tree for that tissue
## similarly, define the lineages to test
## these are the same node names and lineages as from the Create.BAGER.Summary.File.R

######################################################################

## tissue name

tissuename<-"frontal cortex"

## set tissue code

tissue<-"br"

## set paths to directories, be sure to modify your home directory and the Anccestral Reconstruction directory you are analyzing
path="/Users/lynchlab/Desktop/ErinFry/workflowr/AGER/"
pathData=paste(path,"data/forBAGER/",sep="")
pathResults=paste(path,"data/BAGERresults/",tissue,"/",sep="")
pathAncRecon=paste(pathResults,"AncStates/",sep="")

models<-c("VarRates","Lambda")

## set the nodes of interest
if (tissue=="br"){
  node.ancPrimates="Node-00000"
  node.ancHaplorhini="Node-00001"
  node.ancApes="Node-00002"
  node.ancHominini="Node-00003"
  node.ancHomo="Node-00004"
  node.ancPan="Node-00008"
  node.ancPtr="Node-00009"
  node.ancPpa="Node-00016"
  node.ancGorilla="Node-00020"
  node.ancOrang="Node-00023"
  node.ancMacaque="Node-00026"
  
  
} else if (tissue=="cb"){
  node.ancPrimates="Node-00000"
  node.ancHaplorhini="Node-00001"
  node.ancApes="Node-00002"
  node.ancHominini="Node-00003"
  node.ancHomo="Node-00004"
  node.ancPan="Node-00007"
  node.ancPtr="Node-00008"
  node.ancPpa="Node-00011"
  node.ancGorilla="Node-00014"
  node.ancOrang="Node-00017"
  node.ancMacaque="Node-00018"
  
} else if (tissue=="lv"){
  node.ancPrimates="Node-00000"
  node.ancHaplorhini="Node-00001"
  node.ancApes="Node-00002"
  node.ancHominini="Node-00003"
  node.ancHomo="Node-00004"
  node.ancPan="Node-00008"
  node.ancPtr="Node-00009"
  node.ancPpa="Node-00016"
  node.ancGorilla="Node-00020"
  node.ancOrang="Node-00023"
  node.ancMacaque="Node-00026"
  
} else if (tissue=="kd"){
  node.ancPrimates="Node-00000"
  node.ancHaplorhini="Node-00001"
  node.ancApes="Node-00002"
  node.ancHominini="Node-00003"
  node.ancHomo="Node-00004"
  node.ancPan="Node-00008"
  node.ancPtr="Node-00009"
  node.ancPpa="Node-00016"
  node.ancGorilla="Node-00020"
  node.ancOrang="Node-00023"
  node.ancMacaque="Node-00026"
  
} else if (tissue=="ts"){
  node.ancPrimates="Node-00000"
  node.ancApes="Node-00001"
  node.ancHominini="Node-00002"
  node.ancHomo="Node-00003"
  node.ancPan="Node-00006"
  node.ancPtr="Node-00007"
  node.ancPpa="Node-00018"
  node.ancGorilla="Node-00009"
  node.ancMacaque="Node-00010"
  
} else if (tissue=="ht"){
  node.ancPrimates="Node-00000"
  node.ancHaplorhini="Node-00001"
  node.ancApes="Node-00002"
  node.ancHominini="Node-00003"
  node.ancHomo="Node-00004"
  node.ancPan="Node-00008"
  node.ancPtr="Node-00009"
  node.ancPpa="Node-00016"
  node.ancGorilla="Node-00020"
  node.ancOrang="Node-00023"
  node.ancMacaque="Node-00026"
  
} else {
  print("this tissue is not known") } 

## specify each lineage in the tree to test
## formatted that the first column is the ancestral node and the second is the descendent node
lineages.to.test<-matrix(ncol=2, byrow = TRUE,data=c("ancPrimates", "ancMacaque",
                                                     "ancPrimates", "ancHaplorhini",
                                                     "ancHaplorhini", "ancOrang",
                                                     "ancHaplorhini","ancApes",
                                                     "ancApes","ancGorilla",
                                                     "ancApes", "ancHominini",
                                                     "ancHominini","ancPan",
                                                     "ancHominini", "ancHomo",
                                                     "ancPan", "ancPtr",
                                                     "ancPan", "ancPpa"))
```

## Introduction

**This script analyzes the Bayesian Ancestral Gene Expression Reconstructions (BAGERs) to identify expression shifts in the `r tissuename` in all primate lineages. The raw RNA-sequencing files were published by [Brawand et al in 2011](http://www.nature.com/nature/journal/v478/n7369/abs/nature10532.html), collected from post-mortem individuals from 6 primate species. This paper identified only 4 genes with expression shifts in the human frontal cortex using maximum likelihood, but hundreds of expression shifts in the other tissues and lineages. I have quantified RNA-Sequencing raw reads using Ensembl 84, an orthologous exon identification pipeline from Brawand et al, and the psueodaligner Kallisto. I then controlled for quality and confounding variables by regressing out technical variables. [Specifics can be found here](https://erinfry6.github.io/BAGER/PrepDataforBAGER.html).**

**I then reconstructed the ancestral transcriptomes of the ancestral primate nodes using a Bayesian MCMC method implemented in [BayesTraits](http://www.evolution.rdg.ac.uk/BayesTraitsV3/BayesTraitsV3.html).	The evolution of gene expression has not yet been modeled using Bayesian MCMC programs, which model discrete and continuous trait evolution across phylogenies. BayesTraits utilizes a Brownian motion model of trait evolution, a phylogeny, continuous trait data, and evolutionary rate parameters to infer posterior probability distributions of evolutionary rate parameters, and thus ancestral trait values. I have utilized a novel approach implemented in BayesTraits to reconstruct gene expression in the frontal cortex and five other tissues. In each tissue and one gene at a time, my pipeline utilizes transcript abundances, the sample phylogeny, and a MCMC model including a variable rates parameter to reconstruct the evolutionary history of the trait while allowing the rate of change of each geneâ€™s expression to differ across the tree, the most biologically realistic assumption. The algorithm infers evolutionary rate parameters and posterior probability distributions of the ancestral gene expression at each node. [Scripts to run BAGER can be found here](https://github.com/erinfry6/AncGeneExpReconstructionBT).**

**In this document, I identify genes with expression shifts in every by calculating the proportion of MCMC iterations in which there was an increase or decrease in expression in the human lineage (termed the Bayesian Posterior Probability of Divergence, or BPPD).**

## Setup
###Load paths and libraries
```{r set up paths and load libraries, warning=FALSE, message=FALSE}
## Models evaluated in BAGER analysis
print(models)

## set strings as factors as false for everything to run smoothly
options(stringsAsFactors = FALSE)

## load necessary packages
library(biomartr)
library(dplyr)
library("ggplot2")
library(gplots)
library(plotly)
library(WebGestaltR)
```

### Define useful functions
```{r define functions}
## define function for reading in these types of files which will read incorrectly using standard read.csv
read.tcsv = function(file, header=TRUE, sep=",", ...) {
  
  n = max(count.fields(file, sep=sep), na.rm=TRUE)
  x = readLines(file)
  
  .splitvar = function(x, sep, n) {
    var = unlist(strsplit(x, split=sep))
    length(var) = n
    return(var)
  }
  
  x = do.call(cbind, lapply(x, .splitvar, sep=sep, n=n))
  x = apply(x, 1, paste, collapse=sep) 
  out = read.csv(text=x, sep=sep, header=header, ...)
  return(out)
  
}

## set function for importing the ancestral expression reconstructions
read.AncRecon=function(file, firstrow, lastrow,header=F, sep='\t'){
  temp<-t(read.tcsv(file, sep='\t', header=F)[,firstrow:lastrow])
  colnames(temp)<-temp[1,]
  temp<-temp[-1,]
  return(temp)
}

## prepare df or matrix for heatmap.2
prepforheatmap=function(matrix.or.df){
  dims <- dim(matrix.or.df)
matrix.or.df <- as.numeric(matrix.or.df)
dim(matrix.or.df) <- dims 
return(matrix.or.df)
}

## view the posterior probabilty distributions for each gene in a list of genes
viewPPDs<-function(genelist, m=Summary$modelchoice,nbins=100){
  for (i in genelist){  #for each of these genes
  gene<- read.AncRecon(paste(pathAncRecon,m[i],"/",listcsv[i], sep=""), firstrow = it.begin, lastrow = (expectedrows+it.begin), sep='\t') # read reconstruction
    
  anc1<-as.numeric(gene[,which(colnames(gene)==paste(ancHomo, " - 1",sep=""))])
  anc2<-as.numeric(gene[,which(colnames(gene)==paste(ancHominini, " - 1",sep=""))])
  
  minimum=(min(anc1, anc2)) #minimum estimated value
    maximum=(max(anc1, anc2)) #maximum estimated
    breakby= (maximum-minimum)/nbins  #size of bins
    bins <- seq(minimum, maximum, by =breakby)  #set the bins for this gene
  hist(anc1,
       main= paste("Gene #", i,"-", Summary$hgnc_symbol[i],"- BPPD:",Summary$BayesianPostProbofDivergence[i]),  #title of the image
       xlab="Inferred Expression in log(TPM)", 
       ylab="Number of Samples", 
       col="#1c9099", breaks=bins, xlim=c(minimum,maximum))
  hist(anc2, add=T, col="#737373", breaks=bins)
  hist(anc1, border="black", breaks=bins, add=T)
  #legend(300,300, c("Human-Chimpanzee", "Human"), col=c("#7fbf7b", "#af8dc3"), lwd=10, border=NULL)
  } }


## view the posterior divergences along the lineage
viewDifference<-function(genelist,m=Summary$modelchoice, lastit=1000,nbins=100){
  for (i in genelist){  #for each of these genes
  gene<- read.AncRecon(paste(pathAncRecon,m[i],"/",listcsv[i], sep=""), firstrow = it.begin, lastrow = (expectedrows+it.begin), sep='\t')# read reconstruction
  anc1<-as.numeric(gene[,which(colnames(gene)==paste(ancHomo, " - 1",sep=""))])
  anc2<-as.numeric(gene[,which(colnames(gene)==paste(ancHominini, " - 1",sep=""))])
  diffvector<-anc1-anc2
  hist(diffvector,
       main= paste("Gene #", i,"-", Summary$hgnc_symbol[i],"-Post Prob Divergence:",abs(max(1-Summary$BayesianPostProbofDivergence[i], Summary$BayesianPostProbofDivergence[i]))),  #title of the image
       xlab="AncHuman - AncHC (log(TPM))", 
       ylab="Number of Samples", col="#1c9099")
  } }

```

### Load and Examine the BAGER summary file

```{r Load BAGER summary file, echo=FALSE, tidy=TRUE, cache=TRUE}
## uploaded saved file 
Summary<-read.table(paste(pathResults,"2017-07-28BAGERSummary.txt", sep=""),sep='\t', header=T)

Summary[1:5,1:10]

## define the list of genes
listcsv<-Summary$listcsv

## find which row the iteration information begins on for this tissue's tree
finding.information.about.file<-(read.tcsv(paste(pathAncRecon,models[length(models)],"/",listcsv[1],sep=""), sep='\t'))
it.begin<-which(colnames(finding.information.about.file)=="Itter")

## set the total number of rows you expect to have in each file so the code will warn you if reconstructions failed
expectedrows=ncol(finding.information.about.file)-it.begin

```

### Select genes that are expressed with successful reconstructions
Genes that succesfully reconstructed will have similar posterior variations in the two ancestral node reconstructions. Thus, to eliminate successfully reconstructed genes, you calculate the fold difference in standard deivation between the two reconstructions.

```{r find reconstructed genes in each lineage, warnings=FALSE}
## the more ancestral reconstruction generally has a higher standard deviation, but if too large the chain failed to converge on a reconstruction

## define divergence fold standard deviation cutoff
cutoffSD<-4

## only select genes which are expressed, set cutoff for expression
## 2 TPM is a standard expression cutoff
expcut<-log(2)

## filter genes that did not converge or are not expressed in either reconstruction for every lineage
## define reconstructed genes list
reconsgenes<-list()
## for each lineage, select only expressed and reconstructed genes
## then save those genes in the list, but only including relavent statistics for that lineage
x=1
for (l in 1:nrow(lineages.to.test)){
    lineage<-paste(lineages.to.test[l,1],lineages.to.test[l,2], sep="_")
    
    temp<-filter(Summary, eval(as.symbol(paste(lineage,"foldSD",sep="_"))) <cutoffSD,  eval(as.symbol(paste(lineage,"reconAnc",sep="_")))>expcut | eval(as.symbol(paste(lineage,"reconDesc",sep="_")))>expcut)
    
    reconsgenes[[x]]<-temp[,c(1:5,grep(pattern = lineage, x = colnames(temp)))]
    
    x=x+1
  }

## collect summary information for each lineage
num.recons<-sapply(reconsgenes, nrow)

```

**`r (length(which(Summary$foldSD<cutoffSD))/nrow(Summary))*100`% of genes successfully reconstructed (with a cutoff of `r cutoffSD`x standard deviation fold change).**

**`r nrow(reconsgenes)` genes successfully reconstructed and are expressed.**


## Identify genes with expression shifts in the human lineage

**I have chosen to use the Bayesian Posterior Probability of Divergence to identify genes with different expression levels in ancHuman and ancHuman-Chimpanzee nodes. We will use a BPPD of 90% as a cutoff.**

```{r transcriptome qualities of converged reconstructions, echo=FALSE, warning=FALSE}
## define the BPPD cutoff to call a gene to have an expression shift
BPPDcutoff<-0.9


shiftanalysis<-function(reconsdf, BPPDcutoff=0.9, minexpression= expcut, lineage.df){
  
  
  write.table(reconsdf,paste(pathResults,Sys.Date(),lineage.df,"Reconstructed.txt", sep=""),sep='\t')

  
  ## identify all expressed, converged genes with expression shifts
  shiftgenes<-filter(as.data.frame(reconsdf),eval(as.symbol(paste(lineage.df,"BPPD",sep="_")))>BPPDcutoff)

  write.table(shiftgenes,paste(pathResults,Sys.Date(),lineage.df,"90Cutoff.txt", sep=""),sep='\t') 

  
  ## View the distribution of divergence of converged genes
  hist(as.data.frame(reconsdf)[,grep(colnames(reconsdf), pattern = "BPPD")], #create histogram of percent divergence for tested genes
       main= paste("Transcriptome Divergence for", tissuename, lineage, sep=" "), #title
       xlab="Bayesian Posterior Probability of Divergence", 
       ylab="Number of Genes", cex.lab=1.3,
       col="#1c9099", breaks=100)
  abline(v=BPPDcutoff, col="red")

  
  return(c(nrow(reconsdf),nrow(shiftgenes), length(listcsv)))
}




BAGERshiftsummary<-matrix(ncol=nrow(lineages.to.test), nrow=3)
rownames(BAGERshiftsummary)<- c(paste('BPPD>',BPPDcutoff,'%',sep=""),paste('BPPD<',BPPDcutoff,'%',sep=""), "Genes Tested")
colnames(BAGERshiftsummary)<-apply(lineages.to.test, 1, FUN = function(x){paste(x[1],x[2])})

for (l in 1:nrow(lineages.to.test)){
  
BAGERshiftsummary[,l]<-shiftanalysis(reconsgenes[[l]], lineage.df = paste(lineages.to.test[l,1],lineages.to.test[l,2], sep="_")) }

write.table(BAGERshiftsummary,paste(pathResults,Sys.Date(),"BAGERshiftsummary.txt", sep=""),sep='\t')

```





```{r, evaluate=FALSE}
## next look at scatter plot of reconstructions by the Posterior Probability of Divergence (the preferred measure of divergence)
# define divergence cutoffs
reconsdf$BPPDLevel <- cut(reconsdf[,grep(colnames(reconsdf), pattern = "BPPD")], c(0,0.90,0.95,0.975,1.1))

BPPDcats<-reconsdf %>% 
  group_by(BPPDLevel) %>%
  summarise(no_genes=length(BPPDLevel))
BPPDcats

# Reverse the levels and generate some labels for the legend
reconsdf$labels <- factor(reconsdf$BPPDLevel, levels = rev(levels(reconsdf$BPPDLevel)),
                           labels = c('>97.5%', '95-97.5%',
                                       '90-95%','<90%'))
## split the data into layers
df_layer_1 <- reconsdf[ reconsdf$labels =="<90%",]
df_layer_90 <- reconsdf[ reconsdf$labels =="90-95%",]
df_layer_95 <- reconsdf[ reconsdf$labels =="95-97.5%",]
df_layer_99 <- reconsdf[ reconsdf$labels ==">97.5%",]


p<-ggplot(data=reconsdf, mapping=aes(x=(MedianAncHomo),y=(MedianAncHominini), text=hgnc_symbol)) + 
  theme_bw() +
  theme(plot.title= element_text(size=15, face="bold"), axis.title.x=element_text(size=20),axis.title.y=element_text(size=20)) +
  ggtitle("Ancestral Transcript Levels and BBPD in the Human Lineage") +
  labs( x="Ancestral Human-Chimpanzee log(TPM)", y="Ancestral Human log(TPM)", face="bold", size=20) +
  geom_point(data=df_layer_1, colour="#ffffcc", alpha=1) +
  geom_point(data=df_layer_90, colour="#a1dab4", alpha=1) +
  geom_point(data=df_layer_95, colour="#41b6c4", alpha=1) +
  geom_point(data=df_layer_99, colour="#0c2c84", alpha=1) +
  geom_abline(intercept=0,slope=1) + 
  scale_y_log10(limits=c(1e-1,1e2)) + scale_x_log10(limits=c(1e-1,1e2)) +
  geom_vline(xintercept=minexpression, alpha=.5) + geom_hline(yintercept=minexpression, alpha=0.5) 

ggsave(paste(pathResults,"Scatter.pdf"))

ggplotly(p)
}
```

**`r nrow(shiftgenes)` genes were identified to have expression shifts in the human lineage with a Bayesian Posterior Probability of Divergence of 90% of greater.**


## Examine the ancestral gene expression reconstructions...

To visiualize expression shifts, one can view the posterior probability divergences of the ancHuman (teal) and ancHC (gray).


### of genes identified to have an expression shift in frontal cortex from Brawand et al

```{r ID expression shifts and see how fits expectations}

## examine the divergence of genes identified by Brawand et al
for (g in c("LIX1", "SYT15","CENPT", "THBS4")){
viewPPDs(which(Summary$hgnc_symbol==g))
}

```

Both genes identified by Brawand et al in our dataset were identified using our method. 


### of housekeeping genes

We do not expect expression shifts in housekeeping genes.

```{r houskeeping genes}
## examine housekeeping gene divergence
## these are some housekeeping genes
housekeepingBPPD<-vector()
for (g in c("ATF1", "ATF2","ATF4", "BTF3","BAT1","HNRPK","EIF1","EIF1B","EIF2A","EIF4H","AARS","CARS","HARS","RPL8","RPS5","MRPL1","MRPL24","POLR1C","POLR1E","PFDN5","SNX5","H1FX","H2AFX","VCP")){
viewPPDs(which(Summary$hgnc_symbol==g))
  housekeepingBPPD[g]<-max(Summary$BayesianPostProbofDivergence[which(Summary$hgnc_symbol==g)], 0)
}

## what is the distribution of divergences of the housekeeping genes in this list
hist(housekeepingBPPD, breaks = 100, main="Distribution of Divergence of Housekeeping Genes", xlim=c(0,1))
abline(v=0.9)
```
**`r length(which(housekeepingBPPD>0.9))` Housekeeping genes of `r length(housekeepingBPPD)` tested were identified to have expression shifts, with a mean of `r mean(housekeepingBPPD)` BPPD.**


### of newly identified genes with expression shifts using our method

```{r view newly identified shifts}
## view PPDs of genes identified with expression shifts
viewPPDs(which(Summary$hgnc_symbol %in% shiftgenes$hgnc_symbol)[1:10])
viewDifference(which(Summary$hgnc_symbol %in% shiftgenes$hgnc_symbol)[1:10])
``` 


## View original gene expression heatmaps of genes with expression shifts compared to one of all reconstructed genes

These heatmaps are generated using the expression data in all sampled primate frontal cortices. The first, of all reconstructed genes, does not elucidate any obvious patterns. The second, of all genes with expression shifts, shows clusters of genes with specific patterns. The most notable of which is either increased or decreased expression in humans compared to all other species.

```{r examine qualities of genes identified to have shifts in the original data, echo=FALSE, cache=TRUE}
## load expression data used to run scripts
#TPM_log<-t(read.table("TPM_log2.txt",header=F))
#colnames(TPM_log)<-TPM_log[1,]
#TPM_log<-TPM_log[-1,]
## to get it in the right format, save and reload
#write.table(TPM_log, "temp.txt")
TPM_log<-t(as.matrix(read.table(paste(pathData,tissue,"_exp.txt", sep=""),header=F)))

colnames(TPM_log)<-TPM_log[1,]
TPM_log<-TPM_log[-1,]

## select only converged genes to create heatmap
converged_expression<-prepforheatmap(TPM_log[(which(Summary$hgnc_symbol %in% reconsgenes$hgnc_symbol)-1),])

## order of the columns in the heatmaps
colnames(TPM_log)

## set heatmap title size
par(cex.main=.8)

heatmap.2(converged_expression, col=bluered(75),
          density.info="none", trace="none", symm=FALSE,
          dendrogram="row", scale="row",
          Rowv=T, Colv=F,colsep = 1:74, 
          sepcolor="black", 
          sepwidth=c(0.0000000001,0.000000001),
          labRow="", labCol="", main="Raw Expression of Reconstructed Genes")


## select only genes identified to have expression shifts

shift_expression<-prepforheatmap(TPM_log[(which(Summary$hgnc_symbol %in% shiftgenes$hgnc_symbol)-1),])


heatmap.2(shift_expression, col=bluered(75),
          density.info="none", trace="none", 
          dendrogram="row", scale="row",
          Rowv=T, Colv=F,colsep = 1:74, 
          sepcolor="black", 
          sepwidth=c(0.0000000001,0.000000001),
          labRow="", labCol="", main="Expression of all Genes with Expression Shifts")

```

### Misc exploratory analysis
```{r ancrecon as mean exploration, warning=FALSE}
## human mean
human_mean<-apply(converged_expression[,grep(pattern = "hsa", colnames(TPM_log))], 1, mean)

## chimp mean
chimp_mean<-apply(converged_expression[,c(grep(pattern = "ptr", colnames(TPM_log)) , grep(pattern = "ppa", colnames(TPM_log)) )], 1, mean)

## human and chimp mean
HC_mean<-apply(converged_expression[,c(grep(pattern = "hsa", colnames(TPM_log)), grep(pattern = "ptr", colnames(TPM_log)) , grep(pattern = "ppa", colnames(TPM_log)) )], 1, mean)

non_human_mean<-apply(converged_expression[,-grep(pattern = "hsa", colnames(TPM_log))], 1, mean)

temp<-cbind(reconsgenes,human_mean, chimp_mean, HC_mean, non_human_mean)


df_layer_90 <- temp[ temp$labels =="90-95%",]
df_layer_95 <- temp[ temp$labels =="95-99%",]
df_layer_99 <- temp[ temp$labels ==">99%",]



## If BT is averaging the descendants to reconstuct the ancestor, we should see 1:1 between descendant mean and reconstructed mean

## of humans

ggplot(data=temp, mapping=aes(x=(MedianAncHomo),y=(human_mean), text=hgnc_symbol)) + 
  theme_bw() +
  theme(plot.title= element_text(size=15, face="bold"), axis.title.x=element_text(size=20),axis.title.y=element_text(size=20)) +
  ggtitle("Ancestral Transcript Levels and by BBPD in the Human Lineage") +
  labs( x="MedianAncHomo", y="human_mean", face="bold", size=20) +
  geom_point(colour="grey",alpha=0.5)+
    geom_point(data=df_layer_90, colour="#a1dab4", alpha=1) +
  geom_point(data=df_layer_95, colour="#41b6c4", alpha=1) +
  geom_point(data=df_layer_99, colour="#0c2c84", alpha=1) +
  scale_y_log10(limits=c(1e-1,1e2)) + scale_x_log10(limits=c(1e-1,1e2)) +
  geom_abline(intercept=0,slope=1) + 
  geom_vline(xintercept=expcut, alpha=1) + geom_hline(yintercept=expcut, alpha=1) 

## of humans and chimps

ggplot(data=temp, mapping=aes(x=(MedianAncHominini),y=(HC_mean), text=BayesianPostProbofDivergence)) + 
  theme_bw() +
  theme(plot.title= element_text(size=15, face="bold"), axis.title.x=element_text(size=20),axis.title.y=element_text(size=20)) +
  ggtitle("Ancestral Transcript Levels and by BBPD in the Human Lineage") +
  labs( x="MedianAncHominini", y="HC_mean", face="bold", size=20) +
  geom_point(colour="grey",alpha=0.5)+
    geom_point(data=df_layer_90, colour="#a1dab4", alpha=1) +
  geom_point(data=df_layer_95, colour="#41b6c4", alpha=1) +
  geom_point(data=df_layer_99, colour="#0c2c84", alpha=1) +
    scale_y_log10(limits=c(1e-1,1e2)) + scale_x_log10(limits=c(1e-1,1e2)) +
  geom_abline(intercept=0,slope=1) + 
  geom_vline(xintercept=expcut, alpha=1) + geom_hline(yintercept=expcut, alpha=1) 

## the reconstruction of humans is pretty much the mean. Of humans and chipms, there is some variation, but still many are the mean


## Another question: does the method ID genes with different gene expression in humans?

## look at only shift genes comparing human mean to mean of all non_humans

ggplot(data=temp, mapping=aes(x=(human_mean),y=(non_human_mean), text=hgnc_symbol)) + 
  theme_bw() +
  theme(plot.title= element_text(size=15, face="bold"), axis.title.x=element_text(size=20),axis.title.y=element_text(size=20)) +
  ggtitle("Ancestral Transcript Levels and by BBPD in the Human Lineage") +
  labs( x="human_mean", y="non_human_mean", face="bold", size=20) +
  geom_point(colour="grey",alpha=0.5)+
    geom_point(data=df_layer_90, colour="#a1dab4", alpha=1) +
  geom_point(data=df_layer_95, colour="#41b6c4", alpha=1) +
  geom_point(data=df_layer_99, colour="#0c2c84", alpha=1) +
    scale_y_log10(limits=c(1e-1,1e2)) + scale_x_log10(limits=c(1e-1,1e2)) +
  geom_abline(intercept=0,slope=1) + 
  geom_vline(xintercept=expcut, alpha=1) + geom_hline(yintercept=expcut, alpha=1) 

## it is identifying genes with differing means in the original dataset, and giving us a way to model the gene's evolution


## if the reconstructions are basically equal to the mean, we should see the humand and HC mean's diverging for genes with shifts

ggplot(data=temp, mapping=aes(x=(human_mean),y=(HC_mean), text=hgnc_symbol)) + 
  theme_bw() +
  theme(plot.title= element_text(size=15, face="bold"), axis.title.x=element_text(size=20),axis.title.y=element_text(size=20)) +
  ggtitle("Ancestral Transcript Levels and by BBPD in the Human Lineage") +
  labs( x="human_mean", y="HC_mean", face="bold", size=20) +
  geom_point(colour="grey",alpha=0.5)+
    geom_point(data=df_layer_90, colour="#a1dab4", alpha=1) +
  geom_point(data=df_layer_95, colour="#41b6c4", alpha=1) +
  geom_point(data=df_layer_99, colour="#0c2c84", alpha=1) +
    scale_y_log10(limits=c(1e-3,1e2)) + scale_x_log10(limits=c(1e-3,1e2)) +
  geom_abline(intercept=0,slope=1) + 
  geom_vline(xintercept=expcut, alpha=1) + geom_hline(yintercept=expcut, alpha=1) 

##this signal coming from the chimpanzees

ggplot(data=temp, mapping=aes(x=(human_mean),y=(chimp_mean), text=hgnc_symbol)) + 
  theme_bw() +
  theme(plot.title= element_text(size=15, face="bold"), axis.title.x=element_text(size=20),axis.title.y=element_text(size=20)) +
  ggtitle("Ancestral Transcript Levels and by BBPD in the Human Lineage") +
  labs( x="human_mean", y="chimp_mean", face="bold", size=20) +
  geom_point(colour="grey",alpha=0.5)+
    geom_point(data=df_layer_90, colour="#a1dab4", alpha=1) +
  geom_point(data=df_layer_95, colour="#41b6c4", alpha=1) +
  geom_point(data=df_layer_99, colour="#0c2c84", alpha=1) +
    scale_y_log10(limits=c(1e-3,1e2)) + scale_x_log10(limits=c(1e-3,1e2)) +
  geom_abline(intercept=0,slope=1) + 
  geom_vline(xintercept=expcut, alpha=1) + geom_hline(yintercept=expcut, alpha=1) 

```


```{r enrichment analysis, evaluate=FALSE}
## webgestalt R for GO enrichment analysis, see manual https://cran.r-project.org/web/packages/WebGestaltR/WebGestaltR.pdf 

for (enrichdb in c("geneontology_Biological_Process", "geneontology_Cellular_Component","pathway_Panther","pathway_KEGG", "pathway_Wikipathway", "phenotype_Human_Phenotype_Ontology")){

WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = enrichdb,
            enrichDatabaseFile=NULL, enrichDatabaseType=NULL,
            interestGene=shiftgenes$ensembl_gene_id,interestGeneType="ensembl_gene_id",
            collapseMethod="mean",
            referenceGene=reconsgenes$ensembl_gene_id,referenceGeneType="ensembl_gene_id",
            minNum=10, maxNum=500,
            fdrMethod="BH",sigMethod="fdr",fdrThr=0.05,
            topThr=10,dNum=20,perNum=1000,
            lNum=20,is.output=TRUE,
            outputDirectory=paste(pathResults,sep=""),projectName=paste(Sys.Date(),enrichdb, sep="-"),
            keepGSEAFolder=FALSE, methodType="R",dagColor="binary",hostName="http://www.webgestalt.org/")
}
```

